pkg/js/tests/modules/module-to-model.test.ts

```diff
@@ -21,9 +21,14 @@ describe("transformModuleFilesToModel", () => {
                     const errorsCount = testCase.expected_errors.length;
                     expect(exception.message).toEqual(
                         `${errorsCount} error${errorsCount === 1 ? "" : "s"} occurred:\n\t* ${testCase.expected_errors
-                        .map((err: BaseError) => {
-                            let msg = "transformation-error error";
-                            if (err?.line) {
+                        .map((err: ModelValidationSingleError|ModuleTransformationSingleError) => {
+                            let errorType = "transformation-error";
+                            if ((err as ModelValidationSingleError).metadata?.errorType) {
+                                errorType = (err as ModelValidationSingleError).metadata!.errorType;
+                            }
+
+                            let msg = `${errorType} error`;
+                            if (err?.line && !err.metadata) {
```

Comment:
If line and no metadata? Is this expected?

Reply:
This is working around the fact that the message for some of the errors will have line/column numbers (errors added during transformation) whereas errors added during validation won't.

It's sniffing from the expected errors as we won't be using the `metadata` field for the errors that have line/column numbers in the message as they will all be `transformation-error`, we can't use `instanceof` checks as these aren't ever actually converted into real errors they just have all the properties of the error in the JSON